server/src/main/java/org/apache/druid/server/coordinator/duty/RunRules.java

```diff
@@ -128,6 +130,18 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
       boolean foundMatchingRule = false;
       for (Rule rule : rules) {
         if (rule.appliesTo(segment, now)) {
+          if (
+              stats.getGlobalStat(
+                  "totalNonPrimaryReplicantsLoaded") >= paramsWithReplicationManager.getCoordinatorDynamicConfig()
+                                                                                   .getMaxNonPrimaryReplicantsToLoad()
+              && !paramsWithReplicationManager.getReplicationManager().isLoadPrimaryReplicantsOnly()
+          ) {
+            log.info(
+                "Maximum number of non-primary replicants [%d] have been loaded for the current RunRules execution. Only loading primary replicants from here on.",
```

Comment:
Since this behavior is valid only for the present coordinator run, the log message might be clearer with something like "Only loading primary replicants from here on for this coordinator run period"